src/extensions/windows/Microsoft.DotNet.UpgradeAssistant.Extensions.Windows/UWPtoWinAppSDKUpgrade/WinUIAppWIndowAnalyzer.cs

```diff
@@ -0,0 +1,184 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.DotNet.UpgradeAssistant.Extensions.Windows.UWPtoWinAppSDKUpgrade;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Windows
+{
+    [ApplicableComponents(ProjectComponents.WinUI)]
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class WinUIAppWindowAnalyzer : DiagnosticAnalyzer
+    {
+        public const string DiagnosticIdAppWindowNamespace = "UA315_A";
+        public const string DiagnosticIdAppWindowMember = "UA315_B";
+        public const string DiagnosticIdAppWindowVarType = "UA315_C";
+
+        private const string Category = "Upgrade";
+
+        private static readonly LocalizableString Title = "Windows App SDK apps should use Microsoft.UI.Windowing.AppWindow";
+        private static readonly LocalizableString MessageFormat = "'{0}' should use Microsoft.UI.Windowing.AppWindow";
+        private static readonly LocalizableString Description = "Tries to detect the use of Windows.UI.WindowManagement.AppWindow apis";
+
+        private static readonly DiagnosticDescriptor RuleAppWindowNamespace = new(DiagnosticIdAppWindowNamespace, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+        private static readonly DiagnosticDescriptor RuleAppWindowVarType = new(DiagnosticIdAppWindowVarType, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+        private static readonly DiagnosticDescriptor RuleAppWindowMember = new(DiagnosticIdAppWindowMember, Title, MessageFormat, Category, DiagnosticSeverity.Warning, isEnabledByDefault: true, description: Description);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(RuleAppWindowNamespace, RuleAppWindowMember, RuleAppWindowVarType);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            if (context is null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze);
+            context.RegisterSyntaxNodeAction(AnalyzeInvocationExpression, SyntaxKind.InvocationExpression);
+            context.RegisterSyntaxNodeAction(AnalyzeIdentifierName, SyntaxKind.IdentifierName);
+        }
+
+        public static readonly Dictionary<string, KeyValuePair<(string FromNamespace, string FromType), (string ToNamespace, string ToType)>> TypeConversions
```

Comment:
This data object structure seems to be used in a few places. Can you create your own type that isn't a bunch of nested generics and tuples?

Reply:
I am planning on doing that in the upcoming versions
The shape of this data structure has been changing a lot and that's why I didn't decide on doing it for this PR